src/rules/no-multiple-h1.js

```diff
@@ -0,0 +1,93 @@
+/**
+ * @fileoverview Rule to enforce at most one h1 heading in Markdown.
+ * @author Pixel998
+ */
+
+//-----------------------------------------------------------------------------
+// Type Definitions
+//-----------------------------------------------------------------------------
+
+/**
+ * @typedef {import("../types.ts").MarkdownRuleDefinition<{ RuleOptions: [{ frontmatterTitle?: string; }]; }>}
+ * NoMultipleH1RuleDefinition
+ */
+
+//-----------------------------------------------------------------------------
+// Rule Definition
+//-----------------------------------------------------------------------------
+
+/** @type {NoMultipleH1RuleDefinition} */
+export default {
+	meta: {
+		type: "problem",
+
+		docs: {
+			recommended: true,
+			description: "Disallow multiple h1 headings in the same document",
+			url: "https://github.com/eslint/markdown/blob/main/docs/rules/no-multiple-h1.md",
+		},
+
+		messages: {
+			multipleH1: "More than one h1 heading found.",
+		},
+
+		schema: [
+			{
+				type: "object",
+				properties: {
+					frontmatterTitle: {
+						type: "string",
+					},
+				},
+				additionalProperties: false,
+			},
+		],
+
+		defaultOptions: [{ frontmatterTitle: "\\s*title\\s*[:=]" }],
+	},
+
+	create(context) {
+		const [{ frontmatterTitle }] = context.options;
+		const titlePattern =
+			frontmatterTitle === "" ? null : new RegExp(frontmatterTitle, "iu");
+		let h1Count = 0;
+
+		return {
+			yaml(node) {
+				if (titlePattern?.test(node.value)) {
+					h1Count++;
+				}
+			},
+
+			toml(node) {
+				if (titlePattern?.test(node.value)) {
+					h1Count++;
+				}
+			},
+
+			html(node) {
+				if (/<h1[^>]*>.*?<\/h1>/iu.test(node.value)) {
```

Comment:
This regex is fragile, we have following false negatives:

```md
# heading 1

<h1
  class="title"
>
This is a multiline H1
</h1>
```

```md
<h1>First Heading</h1>
Some content

<h1
id="second"
>
Second Heading
</h1>
```

Reply:
False positive:

```md
# h1

<!-- <h1> shdghsdg </h1> --> // This is counted as a duplicate h1 
```

Reply:
```suggestion
				if (H1_TAG_REGEX.test(node.value)) {
```

Also let's hoist this rgexp via a constant to avoid creation of new refexp on each node